lib/glueby/wallet/tapyrus_core_wallet_adapter.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+require 'securerandom'
+require 'bigdecimal'
+
+module Glueby
+  class Wallet
+    class TapyrusCoreWalletAdapter < AbstractWalletAdapter
+      WALLET_PREFIX = 'wallet-'.freeze
+      ADDRESS_TYPE = 'legacy'.freeze
+
+      RPC_WALLET_NOT_FOUND_ERROR_CODE = '-18'
+
+      def create_wallet
+        wallet_id = SecureRandom.hex(32)
+        Contract::RPC.client.createwallet(wallet_name(wallet_id))
+        wallet_id
+      end
+
+      # On TapyrusCoreWallet, there are no way to delete real wallet via RPC.
+      # So, here just unload.
+      def delete_wallet(wallet_id)
+        unload_wallet(wallet_id)
+      end
+
+      def load_wallet(wallet_id)
+        Contract::RPC.client.loadwallet(wallet_name(wallet_id))
+      end
+
+      def unload_wallet(wallet_id)
+        Contract::RPC.client.unloadwallet(wallet_name(wallet_id))
+      end
+
+      def wallets
+        Contract::RPC.client.listwallets.map do |wallet_name|
+          match = /#{WALLET_PREFIX}(?<wallet_id>.+)/.match(wallet_name)
+          next unless match
+
+          match[:wallet_id]
+        end.compact
+      end
+
+      def balance(wallet_id, only_finalized = true)
+        switch_wallet(wallet_id) do |client|
+          # min_conf option is always 1 because the getbalance RPC never returns unconfirmed balance,
+          # even if the min_conf option is set as 0.
+          confirmed = tpc_to_tapyrus(client.getbalance('*', 1))
```

Comment:
The meaning is the same, so we can omit the parameters.

```suggestion
          confirmed = tpc_to_tapyrus(client.getbalance)
```

Reply:
Fixed at 6c7c3775b929d93e66992f267227f8585d58f620